src/YieldDistributor.sol

```diff
@@ -170,6 +188,9 @@ contract YieldDistributor is OwnableUpgradeable {
 
             _latestKey = _currentCheckpoint._key;
         }
+        /* It really feels like L170-190 could be unified into a single loop rather than a separate
+        step for calculating just the first chunk of voting power.
+        */
```

Comment:
Suggestion re: this 

```
function getVotingPowerForPeriod(uint256 _start, uint256 _end, address _account) external view returns (uint256) {
    if (_start >= _end) revert StartMustBeBeforeEnd();
    if (_end > block.number) revert EndAfterCurrentBlock();

    uint32 checkpointCount = BREAD.numCheckpoints(_account);
    if (checkpointCount == 0) return 0;

    Checkpoints.Checkpoint208 memory checkpoint = BREAD.checkpoints(_account, 0);
    if (checkpoint._key > _end) return 0;

    uint256 totalVotingPower = 0;
    for (uint32 i = checkpointCount; i > 0;) {
        checkpoint = BREAD.checkpoints(_account, --i);

        if (checkpoint._key <= _end) {
            uint48 effectiveStart = checkpoint._key < _start ? uint48(_start) : checkpoint._key;
            totalVotingPower += checkpoint._value * (_end - effectiveStart);

            if (checkpoint._key <= _start) {
                totalVotingPower -= checkpoint._value * (_start - checkpoint._key);
                break;
            }
            _end = checkpoint._key;
        }
    }

    return totalVotingPower;
}
``` 
Thoughts @ShaNagbaImuru ?

Reply:
1. It might be cleaner to use a `continue` when the `checkpoint._key > end`, to reduce the indent required.
2. I don't think I understand why you need both of calculating and using `effectiveStart` and subtracting voting power when `checkpoint._key <= _start`. That seems redundant unless I'm missing something. (If it is redundant, I prefer the first approach for clarity.)
3. I think there's probably already unit testing for this code, but if not there should be, to ensure that this change in logic doesn't change the results.

Reply:
Ok, opened #84 for this, will get back in touch regarding this